internal/request/graphql/schema/manager.go

```diff
@@ -99,162 +77,22 @@ func (s *SchemaManager) ResolveTypes() error {
-		schemaTypes.DefaultDirective(),
-		schemaTypes.ExplainDirective(explainEnum),
-		schemaTypes.PolicyDirective(),
-		schemaTypes.IndexDirective(orderEnum, indexFieldInput),
-		schemaTypes.PrimaryDirective(),
-		schemaTypes.RelationDirective(),
-		schemaTypes.MaterializedDirective(),
-	}
-}
-
-func inlineArrayTypes() []gql.Type {
-	return []gql.Type{
-		gql.Boolean,
-		gql.Float,
-		gql.Int,
-		gql.String,
-		gql.NewNonNull(gql.Boolean),
-		gql.NewNonNull(gql.Float),
-		gql.NewNonNull(gql.Int),
-		gql.NewNonNull(gql.String),
+	if err != nil {
+		return nil, err
 	}
-}
-
-// default type map includes all the native scalar types
-func defaultTypes(
-	commitObject *gql.Object,
-	commitLinkObject *gql.Object,
-	commitsOrderArg *gql.InputObject,
-	orderEnum *gql.Enum,
-	crdtEnum *gql.Enum,
-	explainEnum *gql.Enum,
-	indexFieldInput *gql.InputObject,
-) []gql.Type {
-	blobScalarType := schemaTypes.BlobScalarType()
-	jsonScalarType := schemaTypes.JSONScalarType()
-
-	idOpBlock := schemaTypes.IDOperatorBlock()
-	intOpBlock := schemaTypes.IntOperatorBlock()
-	floatOpBlock := schemaTypes.FloatOperatorBlock()
-	booleanOpBlock := schemaTypes.BooleanOperatorBlock()
-	stringOpBlock := schemaTypes.StringOperatorBlock()
-	blobOpBlock := schemaTypes.BlobOperatorBlock(blobScalarType)
-	dateTimeOpBlock := schemaTypes.DateTimeOperatorBlock()
-
-	notNullIntOpBlock := schemaTypes.NotNullIntOperatorBlock()
-	notNullFloatOpBlock := schemaTypes.NotNullFloatOperatorBlock()
-	notNullBooleanOpBlock := schemaTypes.NotNullBooleanOperatorBlock()
-	notNullStringOpBlock := schemaTypes.NotNullStringOperatorBlock()
-	notNullBlobOpBlock := schemaTypes.NotNullBlobOperatorBlock(blobScalarType)
-
-	return []gql.Type{
-		// Base Scalar types
-		gql.Boolean,
-		gql.DateTime,
-		gql.Float,
-		gql.ID,
-		gql.Int,
-		gql.String,
-
-		// Custom Scalar types
-		blobScalarType,
-		jsonScalarType,
-
-		// Base Query types
-
-		// Sort/Order enum
-		orderEnum,
-
-		// Filter scalar blocks
-		idOpBlock,
-		intOpBlock,
-		floatOpBlock,
-		booleanOpBlock,
-		stringOpBlock,
-		blobOpBlock,
-		dateTimeOpBlock,
-
-		// Filter non null scalar blocks
-		notNullIntOpBlock,
-		notNullFloatOpBlock,
-		notNullBooleanOpBlock,
-		notNullStringOpBlock,
-		notNullBlobOpBlock,
-
-		// Filter scalar list blocks
-		schemaTypes.IntListOperatorBlock(intOpBlock),
-		schemaTypes.FloatListOperatorBlock(floatOpBlock),
-		schemaTypes.BooleanListOperatorBlock(booleanOpBlock),
-		schemaTypes.StringListOperatorBlock(stringOpBlock),
-
-		// Filter non null scalar list blocks
-		schemaTypes.NotNullIntListOperatorBlock(notNullIntOpBlock),
-		schemaTypes.NotNullFloatListOperatorBlock(notNullFloatOpBlock),
-		schemaTypes.NotNullBooleanListOperatorBlock(notNullBooleanOpBlock),
-		schemaTypes.NotNullStringListOperatorBlock(notNullStringOpBlock),
-
-		commitsOrderArg,
-		commitLinkObject,
-		commitObject,
-
-		crdtEnum,
-		explainEnum,
-
-		indexFieldInput,
+	validation := gql.ValidateDocument(&s.schema, doc, gql.SpecifiedRules)
```

Comment:
thought: I love the near total lack of documentation in the gql-go lib...

todo: This is quite an odd flow, as we appear to be validating the `doc` against a schema that does not yet contain the contents of `doc`.  Can you please document this call quite heavily.

Reply:
I think you have the validation flow backwards Andy.

The ValidateDocument call *is* to validate that the given SDL document is valid relative to the schema defined up till now (eg, first run only the "default" schema is defined). 

This is the correct flow as I understand it. A comment can help but it doesn't seem unusual or out of place to me.

Reply:
> The ValidateDocument call is to validate that the given SDL document is valid relative to the schema defined up till now (eg, first run only the "default" schema is defined).

That's fine, but need to be documented, particularly given that the SDL is partial atm and does not fully represent the changes that will be made to the GQL-schema once encorporated.  There is a lot of GQL stuff that we do/add that will not be validated by this call.

Reply:
Agreed, all the generated stuff after this won't get validated, but I don't think there's any implication at this call site that it's doing anything more than validating the user provided SDL, which needs to be valid on its own before continuing on with any future schema generation stuff.

Reply:
Yes, I'm totally on board with all of that.

The reason I'm asking for documentation is so that the next person who comes across this line (including our future selves) doesn't need to think terribly hard about why it is done here, and why the premature validation is okay here (and potentially come to the wrong conclusion, wasting even more time and thought).

Reply:
I've added documentation that hopefully makes it a bit more clear as to what's happening

Reply:
Looks good, thanks Keenan - should help a fair bit :)